classes/plugins/PubObjectsExportPlugin.inc.php

```diff
@@ -0,0 +1,523 @@
+		return array(
+			EXPORT_STATUS_ANY => __('plugins.importexport.common.status.any'),
+			EXPORT_STATUS_NOT_DEPOSITED => __('plugins.importexport.common.status.notDeposited'),
+			EXPORT_STATUS_MARKEDREGISTERED => __('plugins.importexport.common.status.markedRegistered'),
+			EXPORT_STATUS_REGISTERED => __('plugins.importexport.common.status.registered'),
+		);
+	}
+
+	/**
+	 * Get status actions for the display to the user,
+	 * i.e. links to a web site with more information about the status.
+	 * @param $pubObject
+	 * @return array (string status => link)
+	 */
+	function getStatusActions($pubObject) {
+		return array();
+	}
+
+	/**
+	 * Get actions.
+	 * @param $context Context
+	 * @return array
+	 */
+	function getExportActions($context) {
+		$actions = array(EXPORT_ACTION_EXPORT, EXPORT_ACTION_MARKREGISTERED);
+		if ($this->getSetting($context->getId(), 'username') && $this->getSetting($context->getId(), 'password')) {
+			array_unshift($actions, EXPORT_ACTION_DEPOSIT);
+		}
+		return $actions;
+	}
+
+	/**
+	 * Get action names.
+	 * @return array (string action => string text)
+	 */
+	function getExportActionNames() {
+		return array(
+			EXPORT_ACTION_DEPOSIT => __('plugins.importexport.common.action.register'),
+			EXPORT_ACTION_EXPORT => __('plugins.importexport.common.action.export'),
+			EXPORT_ACTION_MARKREGISTERED => __('plugins.importexport.common.action.markRegistered'),
+		);
+	}
+
+	/**
+	 * Get the plugin ID used as plugin settings prefix.
+	 * @return string
+	 */
+	abstract function getPluginSettingsPrefix();
+
+	/**
+	 * Return the name of the plugin's deployment class.
+	 * @return string
+	 */
+	abstract function getExportDeploymentClassName();
+
+	/**
+	 * Get the XML for selected objects.
+	 * @param $objects mixed Array of or single published article, issue or galley
+	 * @param $filter string
+	 * @param $context Context
+	 * @return string XML document.
+	 */
+	function exportXML($objects, $filter, $context) {
+		$xml = '';
+		$filterDao = DAORegistry::getDAO('FilterDAO');
+		$exportFilters = $filterDao->getObjectsByGroup($filter);
+		assert(count($exportFilters) == 1); // Assert only a single serialization filter
+		$exportFilter = array_shift($exportFilters);
+		$exportDeployment = $this->_instantiateExportDeployment($context);
+		$exportFilter->setDeployment($exportDeployment);
+		libxml_use_internal_errors(true);
+		$exportXml = $exportFilter->execute($objects, true);
+		$xml = $exportXml->saveXml();
+		$errors = array_filter(libxml_get_errors(), create_function('$a', 'return $a->level == LIBXML_ERR_ERROR ||  $a->level == LIBXML_ERR_FATAL;'));
+		if (!empty($errors)) {
+			$this->displayXMLValidationErrors($errors, $xml);
+			fatalError(__('plugins.importexport.common.error.validation'));
+		}
+		return $xml;
+	}
+
+	/**
+	 * Display XML validation errors.
+	 * @param $errors array
+	 * @param $xml string
+	 */
+	function displayXMLValidationErrors($errors, $xml) {
+		echo '<h2>Validation errors:</h2>';
```

Comment:
At some point these should be localized, but OK if that's not done for the 3.0 release.